core-errors/src/lib.rs

```diff
@@ -159,6 +159,10 @@ pub enum MessageError {
     /// An error occurs in attempt to charge gas for dispatch stash hold.
     #[display(fmt = "Not enough gas to hold dispatch message")]
     InsufficientGasForDelayedSending = 399,
+
+    /// An error when the program id is marked as invalid
+    #[display(fmt = "Program id is forbidden")]
+    InvalidProgramId = 400,
```

Comment:
Better stick one wording: either forbidden or invalid.

But I'd omit these changes. Better introduce these changes in a separate PR (this will resolve #2700 )

Reply:
In that issue it mentions a `SyscallUsageError`, should that be returned instead?